lib/charms/tempo_coordinator_k8s/v0/tempo_api.py

```diff
@@ -0,0 +1,219 @@
+"""tempo_api.
+
+This library implements data accessors the tempo-api interface.  The tempo-api interface is used to
+transfer information about an instance of Tempo, such as how to access and uniquely identify it.  Typically, this is
+useful for charms that create a Tempo instance and want other applications to be able to access it.
+
+## Usage
+
+### Requirer
+
+Add the following to your `charmcraft.yaml` or `metadata.yaml`:
+
+```yaml
+requires:
+  tempo-api:
+    # The example below uses the API for when limit=1.  If you need to support multiple related applications, remove
+    # this and use the list-based data accessor method.
+    limit: 1
+    interface: tempo_api
+```
+
+To implement handling this relation:
+
+* observe the relation-changed event for this relation wherever your charm needs to use this data (this relation DOES
+  NOT automatically observe any events)
+* wherever you need access to the data, create a `TempoApiRequirer` instance and use `.get_data()`
+
+An example implementation is:
+
+```python
+class FooCharm(CharmBase):
+    def __init__(self, framework):
+        super().__init__(framework)
+        self.framework.observe(self.on["tempo-api"].relation_changed, self._on_tempo_api_changed)
+
+    def do_something_with_metadata(self):
+        # Get exactly one related application's data, raising if more than one is available
+        tempo_api = TempoApiRequirer(self.model.relations, "tempo-api")
```

Comment:
I'd put this in `__init__` and store it in an instance attr, that's where we usually put all endpoint wrappers

Reply:
related: https://github.com/canonical/tempo-coordinator-k8s-operator/pull/119/files/b66e87feef79e6a11efdf26b9a317a755b428a31#r1969343473

Reply:
done